mergify_engine/branch_updater.py

```diff
@@ -71,18 +85,23 @@ def update(pull, installation_id, method="merge"):
         last_commit_date = [d for d in out.decode("utf8").split("\n")
                             if d.strip()][-1]
 
-        git("fetch", "--quiet", "upstream", pull.g_pull.base.ref,
+        git("fetch", "--quiet", "upstream", base_branch,
             "--shallow-since='%s'" % last_commit_date)
 
-        if method == "merge":
-            git("merge", "--quiet", "upstream/%s" % base_branch, "-m",
-                "Merge branch '%s' into '%s'" % (base_branch, head_branch))
-            git("push", "--quiet", "origin", head_branch)
-        elif method == "rebase":
-            git("rebase", "upstream/%s" % base_branch)
-            git("push", "--quiet", "origin", head_branch, "-f")
-        else:
-            raise RuntimeError("Invalid branch update method")
+        try:
+            _do_update(git, method, base_branch, head_branch)
+        except subprocess.CalledProcessError as e:
+            if "unrelated histories" in e.output:
```

Comment:
I don't think this appears when you use the `rebase` method.

Reply:
I haven't reproduced the issue, so I put the only error message we have.